index.d.ts

```diff
@@ -0,0 +1,19 @@
+type Points = number[][]
+
+declare class KDBush<T> {
+  ids: number[]
+  coords: number[]
+  nodeSize: number
+  points: T[]
+  range(minX: number, minY: number, maxX: number, maxY: number)
+  within(x: number, y: number, r: number)
+}
+
+interface KDBush<T> {
+  (points: Points): KDBush<Points>
+  <T>(points: T[], getX: (p: T) => void, getY: (p: T) => void, nodeSize?: number, ArrayType?: any): KDBush<T>
+}
+
+declare const kdbush: KDBush<Points>
```

Comment:
Can you explain how this works? The templating is a bit confusing... Does it apply to the interface or to the class? What's up with the mismatch between `KDBush<Points>` where `Points` is an array of points, and `KDBush<T>` where `T` is supposed to be an individual point?